src/main/java/io/supertokens/cronjobs/bulkimport/ProcessBulkImportUsers.java

```diff
@@ -0,0 +1,597 @@
+                    // If this update statement fails then the outer transaction will fail as well and the user will simpl be processed again. No inconsistency will happen in this
+                    // case.
+                    baseTenantStorage.updateBulkImportUserPrimaryUserId(appIdentifier, user.id,
+                            primaryLM.superTokensUserId);
+
+                    // We need to commit the transaction manually because we have overridden that in the proxy storage
+                    // If this fails, the primaryUserId will be updated in the bulkImportUser but it wouldn’t actually exist.
+                    // When processing the user again, we'll check if primaryUserId exists with the same email. In this case the user won't exist, and we'll simply re-process it.
+                    bulkImportProxyStorage.commitTransactionForBulkImportProxyStorage();
+
+                    // NOTE: We need to use the baseTenantStorage as bulkImportProxyStorage could have a different storage than the baseTenantStorage
+                    // If this fails, the primaryUserId will be updated in the bulkImportUser and it would exist in the database.
+                    // When processing the user again, we'll check if primaryUserId exists with the same email. In this case the user will exist, and we'll simply delete the entry.
+                    baseTenantStorage.startTransaction(con2 -> {
+                        baseTenantStorage.deleteBulkImportUser_Transaction(appIdentifier, con2, user.id);
+                        return null;
+                    });
+                    return null;
+                } catch (StorageTransactionLogicException e) {
+                    // We need to rollback the transaction manually because we have overridden that in the proxy storage
+                    bulkImportProxyStorage.rollbackTransactionForBulkImportProxyStorage();
+                    throw e;
+                } finally {
+                    closeAllProxyStorages();
+                }
+            });
+        } catch (StorageTransactionLogicException | InvalidBulkImportDataException e) {
+            handleProcessUserExceptions(appIdentifier, user, e, baseTenantStorage);
+        }
+    }
+
+    private void handleProcessUserExceptions(AppIdentifier appIdentifier, BulkImportUser user, Exception e,
+            BulkImportSQLStorage baseTenantStorage)
+            throws StorageQueryException {
+
+        // Java doesn't allow us to reassign local variables inside a lambda expression
+        // so we have to use an array.
+        String[] errorMessage = { e.getMessage() };
+
+        if (e instanceof StorageTransactionLogicException) {
+            StorageTransactionLogicException exception = (StorageTransactionLogicException) e;
+            errorMessage[0] = exception.actualException.getMessage();
```

Comment:
there are many instances where you are wrapping StorageQueryException with StorageTransactionLogicException, and then those will get saved here too. Those should not be saved here, and the record should be left for processing status to be picked up later on. 

Only semantic error resulting from a bad user input should be saved as failed

Reply:
`StorageQueryException` are usually `SQLException`, right?  In most cases, they won't be successful on retries (and it would keep on processing every 10 mins). I think its better to mark it as an error.

Reply:
StorageQueryException usually means something like the db is down or something. All errors that are due to semantic issues have a different type (like emailalreadyexistsexception). Please double check error types

Reply:
We don't mark the user FAILED on `StorageQueryException` anymore. Instead these users will be retried.